### PR TITLE
feat: Change serverless-connector input type and update metadata

### PR DESCRIPTION
**Pull request by q2w**
This PR makes below changes,

1. Updates input variable type for `vpc_connectors` from `list(map(string))` to `list(object({}))`
2. Updates metadata

### CODE REVIEW

File: modules/vpc-serverless-connector-beta/variables.tf [hunk 1]
```diff
@@ -20,7 +20,18 @@ variable "project_id" {
...
 
 variable "vpc_connectors" {
-  type        = list(map(string))
+  type = list(object({
+    name              = string,
+    region            = string,
+    network           = optional(string, null),
+    subnet_name       = optional(string, null),
+    ip_cidr_range     = optional(string, null),
+    host_project_name = optional(string, null),
```
Comment (bharathkkb) [inline modules/vpc-serverless-connector-beta/variables.tf:29]: I think this is the test failure reason
```suggestion
    host_project_id = optional(string, null),
```

Comment (q2w) [reply modules/vpc-serverless-connector-beta/variables.tf]: Done!

File: modules/vpc/metadata.yaml [hunk 2]
```diff
@@ -19,121 +19,146 @@ metadata:
...
-  - name: network_self_link
-    description: The URI of the VPC being created
-  - name: project_id
-    description: VPC project id
-  roles:
-  - level: Project
+  info:
+    title: Terraform VPC Module
+    source:
+      repo: https://github.com/q2w/terraform-google-network.git
```
Comment (bharathkkb) [inline modules/vpc/metadata.yaml:25]: nit: here and a few places
```suggestion
      repo: https://github.com/terraform-google-modules/terraform-google-network.git
```

**Review by bharathkkb — APPROVED** [review]

File: modules/vpc/metadata.yaml (hunk 2 above)
Comment (q2w) [reply modules/vpc/metadata.yaml]: Thanks! Done.

**Pull request merged**